### PR TITLE
[Merged by Bors] - Add AutoMax next to ScalingMode::AutoMin

### PR DESCRIPTION
**Pull request by DasLixou**
# Objective

`ScalingMode::Auto` for cameras only targets min_height and min_width, or as the docs say it `Use minimal possible viewport size while keeping the aspect ratio.`

But there is no ScalingMode that targets max_height and Max_width or `Use maximal possible viewport size while keeping the aspect ratio.`

## Solution

Added `ScalingMode::AutoMax` that does the exact opposite of `ScalingMode::Auto`

---

## Changelog

Renamed `ScalingMode::Auto` to `ScalingMode::AutoMin`.

## Migration Guide

just rename `ScalingMode::Auto` to `ScalingMode::AutoMin` if you are using it.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Great first PR, thanks! I'm not going to merge this for 0.9, to avoid creating more work for the migration guide authors, but I expect this will be on `main` very shortly after.

**Comment by DasLixou** [comment]
Ok thanks for the great and fast feedback :) - Then ill just last the PR here and wait for more comments and problems etc

**Comment by DasLixou** [comment]
I’ve been thinking about also adding an Custom / Closure ScalingMode for implementing own logic. Opinion?

**Comment by alice-i-cecile** [comment]
Might be a good choice, but belongs in its own PR. I'm unsure how useful that would be: feels more complicated and less flexible than being able to dynamically change this in a system with a `Manual` option.

**Comment by DasLixou** [comment]
That would probably be the better solution… ok well than I’ll screw that idea 😅

**Comment by DasLixou** [comment]
So is there something that I still have to do? @alice-i-cecile

**Comment by alice-i-cecile** [comment]
No, we just need another reviewer :) @bzm3r, can you review?

File: crates/bevy_render/src/camera/projection.rs
Comment (bzm3r) [inline crates/bevy_render/src/camera/projection.rs:247]: Hey, can you provide some clarification on why this must be `width * max_height < max_width * height`?

**Review by bzm3r — COMMENTED** [review]
I think this makes sense to me, at least mathematically. I guess I a

File: crates/bevy_render/src/camera/projection.rs
Comment (DasLixou) [reply crates/bevy_render/src/camera/projection.rs:247]: Yeah sure I’ll try to explain it. Will probably be done by tomorrow :)

Comment (DasLixou) [reply crates/bevy_render/src/camera/projection.rs:251]: Nvm just done it. Think this is ok. Or is it still messy? I’m not really good at explaining something in few words. 😅

**Comment by DasLixou** [comment]
But if you want to know if it’s working as expected - I’m using it in my game and it works as it should. :)

File: crates/bevy_render/src/camera/projection.rs
Comment (alice-i-cecile) [inline crates/bevy_render/src/camera/projection.rs:250]:
```suggestion
                // Then use smaller (max_height when true) as what it referres to (height when true) and calculate rest so it can't get over maximum.
```

File: crates/bevy_render/src/camera/projection.rs
Comment (alice-i-cecile) [inline crates/bevy_render/src/camera/projection.rs:238]:
```suggestion
                // Then use bigger (min_height when true) as what it refers to (height when true) and calculate rest so it can't get under minimum.
```

**Review by bzm3r — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
There was a mirror typo in the other comment that needs to be fixed too :)

**Comment by DasLixou** [comment]
> There was a mirror typo in the other comment that needs to be fixed too :)

Whoops - I’ll fix it :) - isn’t that simple with a German autocorrection and a smartphone 😅

**Comment by DasLixou** [comment]
Now everything should be fine - hopefully 😅

**Comment by alice-i-cecile** [comment]
Awesome! This is now ready to be merged, but because it's a breaking change I'm going to hold off until after 0.9.1 drops to reduce the amount of cherrypicking needed.

**Comment by cart** [comment]
> Awesome! This is now ready to be merged, but because it's a breaking change I'm going to hold off until after 0.9.1 drops to reduce the amount of cherrypicking needed.

I'm cool with merging things now. I'd prefer not to hold back our dev processes for patch releases when cherry picking is straightforward.

**Pull request closed without merging**
